src/main/java/MisterDuke.java

```diff
@@ -0,0 +1,678 @@
+     * @param tasksArray array list of tasks
+     */
+    public void showList(ArrayList<Task> tasksArray) {
+        if (tasksArray.isEmpty()) {
+            System.out.println("     Your list is empty, there is nothing to do. Yay!");
+        } else {
+            System.out.println("     Here are the tasks in your list:");
+            for (int i = 0; i < tasksArray.size(); i++) {
+                System.out.println("       " + (i + 1) + ". " + tasksArray.get(i).toString());
+            }
+        }
+    }
+
+    /**
+     * Informs user that the specified task has been removed from the list
+     * @param tasksArray array list of tasks
+     * @param commandNumber the task in the list that will be removed
+     */
+    public void showTaskDelete(ArrayList<Task> tasksArray, String commandNumber) {
+        if (tasksArray.isEmpty()) {
+            System.out.println("     Oops! You have no tasks to delete.");
+        } else {
+            int cmdNum = Integer.parseInt(commandNumber); //strArray[1]
+            System.out.println("     Noted. I've removed this task: ");
+            System.out.println("       " + tasksArray.get(cmdNum - 1).toString());
+            tasksArray.remove(cmdNum - 1);
+        }
+    }
+
+    /**
+     * Informs the user of wrong/incomplete input
+     * @param e error message that specifies wrong/incomplete input
+     */
+    public void showDefaultError(Exception e) {
+        System.out.println(e.getMessage());
+    }
+
+    public void showOutOfBounds() {
+        System.out.println("     Oops! You don't have that many tasks");
+    }
+
+    public void showMatchingItems(ArrayList<Task> tasksArray) {
+        if (tasksArray.isEmpty()) {
+            System.out.println("     Oh no! There are no matching tasks :(");
+        } else {
+            System.out.println("     Here are the matching tasks in your list:");
+            for (int i = 0; i < tasksArray.size(); i++) {
+                System.out.println("       " + (i + 1) + ". " + tasksArray.get(i).toString());
+            }
+        }
+    }
+}
+
+/**
+ * The Storage class is for loading and saving the list of tasks
+ * to the local hard disk.
+ */
+class Storage {
+    private final File txtFile;
+
+    public Storage (String filePath) throws IOException {
+        this.txtFile = new File(filePath);
+        if (txtFile.createNewFile()) {
+            System.out.println("File created!");
+        } else {
+            System.out.println("File loaded!");
+        }
+    }
+
+    /**
+     * Loads list of tasks from the hard disk into the list of tasks
+     * @return array list of tasks
+     * @throws FileNotFoundException when the file cannot be found on the hard disk
+     */
+    public ArrayList<Task> load() throws FileNotFoundException {
+        Scanner sc = new Scanner(this.txtFile);
+        ArrayList<Task> tasksArrayList = new ArrayList<>();
+
+        while (sc.hasNextLine()) {
+            String tempStr = sc.nextLine();
```

Comment:
I believe it would be better to name your variable fully ("tempString" instead of "tempStr") to improve readability!